### PR TITLE
Log contents of pods on creation/update/deletion in controllers

### PR DESCRIPTION
**Pull request by janetkuo**
Ref #22270; for debugging the test flake 

@kubernetes/sig-config

### CODE REVIEW

**Comment by bgrant0607** [comment]
LGTM

File: pkg/controller/replicaset/replica_set.go [hunk 1]
```diff
@@ -289,6 +289,7 @@ func isReplicaSetMatch(pod *api.Pod, rs *extensions.ReplicaSet) bool {
 // When a pod is created, enqueue the replica set that manages it and update it's expectations.
 func (rsc *ReplicaSetController) addPod(obj interface{}) {
 	pod := obj.(*api.Pod)
+	glog.V(4).Infof("Pod %s created: %+v.", pod.Name, pod)
```
Comment (kargakis) [inline pkg/controller/replicaset/replica_set.go]: Should also happen in the rc controller.

Comment (bprashanth) [inline pkg/controller/replicaset/replica_set.go]: The rc logs pods.labels, do we really need the whole pod? is there something in the annotations we need? if not I'd prefer this also just logged labels to reduce log spam

**Comment by eparis** [comment]
This PR is already included in the release-1.2 branch and does not require cherrypick-ing

**Pull request merged**